crates/oxc_query/src/edges.rs

```diff
@@ -0,0 +1,2170 @@
+            } else {
+                Box::new(std::iter::empty())
+            }
+        })
+    }
+
+    pub(super) fn property<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            Box::new(
+                v.as_class()
+                    .unwrap_or_else(|| {
+                        panic!("expected to have a class vertex, instead have: {v:#?}")
+                    })
+                    .class
+                    .body
+                    .body
+                    .iter()
+                    .filter_map(|class_el| {
+                        if let ClassElement::PropertyDefinition(property) = class_el {
+                            Some(Vertex::ClassProperty(Rc::new(ClassPropertyVertex {
+                                property,
+                                is_abstract: false,
+                            })))
+                        } else if let ClassElement::TSAbstractPropertyDefinition(def) = class_el {
+                            Some(Vertex::ClassProperty(Rc::new(ClassPropertyVertex {
+                                property: &def.property_definition,
+                                is_abstract: true,
+                            })))
+                        } else {
+                            None
+                        }
+                    }),
+            )
+        })
+    }
+
+    pub(super) fn setter<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        class_fn_edge_implem!(contexts, Set)
+    }
+}
+
+pub(super) fn resolve_class_ast_edge<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    edge_name: &str,
+    _parameters: &EdgeParameters,
+    resolve_info: &ResolveEdgeInfo,
+    adapter: &'a Adapter<'b>,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+    match edge_name {
+        "ancestor" => class_ast::ancestor(contexts, resolve_info, adapter),
+        "constructor" => class_ast::constructor(contexts, resolve_info),
+        "entire_class_span" => class_ast::entire_class_span(contexts, resolve_info),
+        "getter" => class_ast::getter(contexts, resolve_info),
+        "method" => class_ast::method(contexts, resolve_info),
+        "name_span" => class_ast::name_span(contexts, resolve_info),
+        "parent" => class_ast::parent(contexts, resolve_info, adapter),
+        "property" => class_ast::property(contexts, resolve_info),
+        "setter" => class_ast::setter(contexts, resolve_info),
+        "span" => class_ast::span(contexts, resolve_info),
+        _ => {
+            unreachable!("attempted to resolve unexpected edge '{edge_name}' on type 'ClassAST'")
+        }
+    }
+}
+
+mod class_ast {
```

Comment:
in reference to earlier when I said we should consider if we need two matching mod's for the same Class vertex, for the AST and non-AST variants. It's a lot more code across every possible AST vertex, but maybe being more verbose is the better choice...

Reply:
Do we expect them to diverge in implementation? If you're sure the answer is "no" then it's worth merging their code. If not, then it's complicated.

Reply:
No, the only change is that the *AST variant adds a few helper edges for traversing ast such as ancestor and parent

Reply:
I decided to leave them for now in edges.rs